ssa/utils.go

```diff
@@ -318,248 +322,12 @@ func AnyInMetadata(object *unstructured.Unstructured, metadata map[string]string
 	return false
 }
 
-// SetNativeKindsDefaults implements workarounds for server-side apply upstream bugs affecting Kubernetes < 1.22
-// ContainerPort missing default TCP proto: https://github.com/kubernetes-sigs/structured-merge-diff/issues/130
-// ServicePort missing default TCP proto: https://github.com/kubernetes/kubernetes/pull/98576
-// PodSpec resources missing int to string conversion for e.g. 'cpu: 2'
-// secret.stringData key replacement add an extra key in the resulting data map: https://github.com/kubernetes/kubernetes/issues/108008
+// SetNativeKindsDefaults sets default values for native Kubernetes objects,
+// working around various upstream Kubernetes API bugs.
 func SetNativeKindsDefaults(objects []*unstructured.Unstructured) error {
```

Comment:
Theoretically, we could introduce a `NormalizeUnstructuredList(objects []*unstructured.Unstructured)` and deprecate this one.

Reply:
With all the changes to follow I think we can go with a major version bump for SSA, so I would rename the function.